### DISCUSSION TITLE
[app beta] How to show the "loading.js" while "page.js" fetches?

### DISCUSSION THREAD

**Discussion by IvoBiaus** [vercel/next.js#48138] [Help] [answered]
### Summary

I have this simple file structure, im trying to simulate the loading.

app/
---`layout.tsx`
---`page.tsx`
---characters/
------`layout.tsx`
------`page.tsx`
------`loading.tsx`

layout.tsx
```tsx 
type Params = {
  children: React.ReactNode;
};

export default function CharactersLayout({ children }: Params) {
  return (
    <div>
      Characters list
      <div>{children}</div>
    </div>
  );
}
```

page.tsx
```tsx
function simulateServerResponse<T>(delay: number, data: T): Promise<T> {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve(data);
    }, delay);
  });
}

export default async function CharactersList() {
  const data = await simulateServerResponse(5000, {
    chars: [1, 2, 3, 4, 5, 6, 7, 8],
  });

  return <div>A list of characters {data.chars}</div>;
}
```

loading.tsx
```tsx
export default function CharactersLoading() {
  return <div>loading...</div>;
}

```

But when going to the `/characters` path, the site starts loading without showing the loading component.
The site changed from `app/page.tsx` directly to `app/characters/page.tsx` skipping the loading.
![image](https://user-images.githubusercontent.com/43274372/230735912-282d6d79-d399-42d8-b8d3-8eb9950a7734.png)

**Comment by IvoBiaus** [comment, marked as answer]
Well i found the fix.
The bug happens when writing the URL by hand.
Apparently it behaves correctly when navigating to the URL with a "Link" component. The `loading.tsx` content shows up inmediatly for 5 seconds and then the `page.tsx`.

**Reply by aureliopita-au** [reply]
I'm also seeing this when using `router.push()`, we get no loading. Is this expected behaviour, only show loading when using`<Link>`?

**Reply by IvoBiaus** [reply]
@aureliopita-au 
I just did a test using `router.push()` (where router is `const router = useRouter()` imported from `next/navigation`) and seems it works correctly, i got the same behaviour as using Link.
- "Page A" with a Link to "Page B", when clicked the link -> inmediatly mounts `loading.tsx` screen for 5 seconds and then mounted the "Page B" screen.
- "Page A" with a Button that does `router.push("/pageB")`, when clicked the Button -> inmediatly mounts `loading.tsx` screen for 5 seconds and then mounted the "Page B" screen.

Its not relevant where you keep your loading file by the way. I tested for example at `/app/pageB/loading.tsx` and `/app/loading.tsx` and both worked.
